app/main.py

```diff
@@ -1,3 +1,58 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (float, int)):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        if isinstance(other, (float, int)):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: float | int) -> Distance:
+        if isinstance(other, (float, int)):
+            return Distance(round(self.km / other, 2))
```

Comment:
other can be Distance too

Reply:
@Bohdan-Salamakha , 
<img width="730" alt="image" src="https://user-images.githubusercontent.com/77115535/226108383-eda95f7a-70dd-4dbf-8c03-98d5b2953186.png">